### PR TITLE
RI-133: Inital MaaS Commit for Designate

### PR DESCRIPTION
**Pull request by leader716**
Initial commit for monitoring designate services the checks include scripts for performing local checks for the API and the designate-mdns service. Created a template for checking designate service processes but not currently creating a check for those processes.

### CODE REVIEW

File: playbooks/files/rax-maas/plugins/designate_api_local_check.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from maas_common import metric
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+import requests
+
+
+def check(auth_ref, args):
+    DESIGNATE_ENDPOINT = 'http://{ip}:9001/'.format(ip=args.ip,)
```
Comment (npawelek) [inline playbooks/files/rax-maas/plugins/designate_api_local_check.py]: We'll need to have an override allowing the endpoint to be built with either http or https.

Comment (leader716) [reply playbooks/files/rax-maas/plugins/designate_api_local_check.py]: This was updated in a new commit

File: playbooks/maas-openstack-designate.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        dest: "/etc/rackspace-monitoring-agent.conf.d/designate_mdns_local_check--{{ inventory_hostname }}.yaml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+      delegate_to: "{{ physical_host | default(ansible_host) }}"
+
+    - name: Install designate lb checks
+      template:
+        src: "templates/rax-maas/lb_api_check_designate.yaml.j2"
+        dest: "/etc/rackspace-monitoring-agent.conf.d/lb_api_check_desugbate.yaml"
```
[the remark this answers is not in the document]
Comment (leader716) [reply playbooks/maas-openstack-designate.yml]: Fixed

File: playbooks/maas-openstack-designate.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        mode: "0644"
+      delegate_to: "{{ physical_host | default(ansible_host) }}"
+      when:
+        - maas_remote_check | bool
+        - not maas_private_monitoring_enabled
+
+    - name: Install designate private lb checks
+      template:
+        src: "templates/rax-maas/private_lb_api_check_designate.yaml.j2"
+        dest: "/etc/rackspace-monitoring-agent.conf.d/private_lb_api_check--{{ inventory_hostname }}.yaml"
```
[the remark this answers is not in the document]
Comment (leader716) [reply playbooks/maas-openstack-designate.yml]: Fixed

**Comment by leader716** [comment]
recheck_cit_xenial_newton_deploy

**Pull request merged**